### ISSUE TITLE
"Malformed response" output 

### ISSUE THREAD

**Issue by mozai** [chubin/wttr.in#199]
$ curl wttr.in/Montreal
2018/05/26 14:39:40 invalid character '<' looking for beginning of value
2018/05/26 14:39:40 Malformed response.

Earlier (closed) issues say this happens when the data source is misbehaving.  "invalid character '<'" looks like an error thrown by an XML parser.
Could catch this error, and send to the user something more appropriate like "could not parse weather data; try again later."

**Comment by chubin** [comment]
Sometimes it happens. It is because of our data backends. That means that one of them was experiencing some problems at that moment. Do you still have this problem?

**Comment by evanniedojadlo** [comment]
Yes, I am still seeing this issue as well.

**Comment by chubin** [comment]
Strange. I can't reproduce it. Are you experiencing this issue for other locations too?

**Comment by evanniedojadlo** [comment]
No, I just retried and was able to curl a couple of different locations, strange.

**Comment by mozai** [comment]
I know you can't prevent the problem; from the other reports, the problem is up-stream from your service.  I'm suggesting a better message to your end-user when you get bad data from upstream. Something like "weather data cannot be parsed at this time, please check again later" when your software sees bad data from upstream.  Could write to your logfile what query prompted the bad data, and how the data was unparsable  (was it a typo? truncated answer? connection time-out? missing elements after parsing?)
